common/generate_elastic_template.py

```diff
@@ -0,0 +1,96 @@
+"""
+Generate an elastic scattering input template for Fresco.
+"""
+
+from fractions import Fraction
+from pathlib import Path
+
+from ._utils import _is_fraction_integer_or_half_integer
+
+
+def load_template(file_path: Path) -> str:
+    with open(file_path, "r") as file:
+        return file.read()
+
+
+template_file_path = Path(__file__).parent / "templates/elastic.template"
+with open(template_file_path, "r") as file:
+    elastic_input_template = file.read()
+
+
+def generate_elastic_template(
+    output_path: Path,
+    mass_t: float,
+    charge_t: float,
+    spin_t: Fraction,
+    mass_p: float,
+    charge_p: float,
+    spin_p: Fraction,
+    E_lab: float,
+    J_tot_min: Fraction,
+    J_tot_max: Fraction,
+    R_Coulomb: float,
+    reaction_name: str,
+    BE_t: float = 0.0,
+    R_match: float = 60.0,
+    step_size: float = 0.1,
+):
+    """
+    Generate an elastic scattering input template for Fresco.
+
+    Parameters:
+    output_path (Path): Path to save the generated template file.
+    mass_t (float): Mass of the target nucleus.
+    charge_t (float): Charge of the target nucleus.
+    spin_t (Fraction): Spin of the target nucleus (integer or half-integer).
+    mass_p (float): Mass of the projectile nucleus.
+    charge_p (float): Charge of the projectile nucleus.
+    spin_p (Fraction): Spin of the projectile nucleus (integer or half-integer).
+    E_lab (float): Laboratory energy of the projectile in MeV.
+    J_tot_min (Fraction): Minimum total angular momentum (integer or
+        half-integer).
+    J_tot_max (Fraction): Maximum total angular momentum (integer or
+        half-integer).
+    R_Coulomb (float): Coulomb radius in fm.
+    reaction_name (str): Name of the reaction for file naming.
+    BE_t (float): Binding energy of the target in MeV. Default is 0.0.
+    R_match (float): Matching radius in fm. Default is 60.0.
+    step_size (float): Step size for the radial mesh in fm. Default is 0
+    """
+
+    if J_tot_min > J_tot_max:
+        raise ValueError("J_tot_min cannot be greater than J_tot_max.")
+    if J_tot_min < 0 or J_tot_max < 0:
+        raise ValueError("J_tot_min and J_tot_max must be non-negative.")
+    if not _is_fraction_integer_or_half_integer(J_tot_min):
+        raise ValueError("J_tot_min must be an integer or half-integer.")
+    if not _is_fraction_integer_or_half_integer(J_tot_max):
+        raise ValueError("J_tot_max must be an integer or half-integer.")
```

Comment:
This function is in the high-level public interface of the package.  We should, therefore, explicitly check all arguments carefully.

If we should do this after closing Issue #29, please add a TODO that states the need and cite that Issue.

Ditto for inelastic.